netci.groovy

```diff
@@ -520,24 +596,91 @@ def static setMachineAffinity(def job, def os, def architecture, def options = n
         assert os != 'Windows_NT'
         assert os in supportedArmLinuxOs
 
-        if (architecture == 'arm' || architecture == 'armlb') {
-            Utilities.setMachineAffinity(job, 'Ubuntu', 'arm-cross-latest')
-        } else {
-            // Arm64 Linux
-            if (options['is_build_only'] == true) {
+        if (architecture == 'arm64') {
+            if ((options != null) && (options['is_build_only'] == true)) {
+                // Arm64 Linux build machine
                 Utilities.setMachineAffinity(job, os, 'arm64-cross-latest')
             } else {
-                // Arm64 Test Machines
-                if (options['large_pages'] == false) {
-                    Utilities.setMachineAffinity(job, os, 'arm64-small-page-size')
-                } else {
+                // Arm64 Linux test machines
+                if ((options != null) && (options['large_pages'] == true)) {
                     Utilities.setMachineAffinity(job, os, 'arm64-huge-page-size')
+                } else {
+                    Utilities.setMachineAffinity(job, os, 'arm64-small-page-size')
+                }
+            }
+        }
+        else if (architecture == 'armem') {
+            // arm emulator (Ubuntu/Ubuntu16.04/Tizen). Build and test on same machine,
+            // using Docker.
+            Utilities.setMachineAffinity(job, 'Ubuntu', 'arm-cross-latest')
```

Comment:
It is worth making sure these jobs need the arm image. I believe the answer is no.

Reply:
I didn't change this. The comments in the agents map say:
```
// Use ubuntu14.04 images
// Contains the rootfs setup for arm/arm64 builds.
```
and:
```
// Contains Mono 5.0.1
```